extras/ams_wrapper/src/api/models/vehicle_detection_adas_model.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import sys
+import json
+from logger import get_logger
+from api.models.model import Model
+
+logger = get_logger(__name__)
+
+
+class VehicleDetectionAdas(Model):
+
+    def load_default_labels(self):
+        labels_path = os.path.abspath(__file__).replace(".py",".json")
+        try:                                                                          
+            with open(labels_path, 'r') as labels_file:                               
+                data = json.load(labels_file)                                         
+                self.labels = data
+        except Exception as e:                                                        
+            logger.exception("Error occurred while opening labels file: {}".format(e))
+            sys.exit(1)                                                               
+        return                                                                   
+
+
+    def postprocess_inference_output(self, inference_output: dict) -> str:
+        from api.types import Tag, Rectangle, SingleEntity, Entity
+
+        result_array = inference_output["detection_out"]
+
+        # model with output shape (1,1,200,7) 
+        # with last dimension containg detection details
+        detection_threshold = 0.5
+        detections = []
+        for detection in result_array[0][0]:
+            label = str(detection[1].item())
+            # End of detections
+            if label == "0.0":
+                break
+
+            if not label in self.labels:
+                label_value = "unknown"
+            else:
+                label_value = self.labels[label]
+
+            image_id = detection[0].item()
+            conf = detection[2].item()
+            x_min = detection[3].item()
+            y_min = detection[4].item()
+            x_max = detection[5].item()
+            y_max = detection[6].item()
+
+            if conf >= detection_threshold:
+                tag = Tag(label_value, conf)
+
+                box = Rectangle(x_min, y_min, abs(x_max-x_min), abs(y_max-y_min))
+
+                detection = SingleEntity(tag, box)
+                detections.append(detection)
+        
```

Comment:
Here you could check if there are any detections at all. If the `detections` list is empty you could raise exception so the processing loop will catch it and return appropriate code. 
This exception should be defined at least on the Model class level, or somewhere else (maybe separate file for errors definitions?).

Reply:
Should we throw exceptions in such cases ? Is it not normal for an image to not detect an object ?

Reply:
Perhaps you're right, but either way we need to handle that situation. If not with exceptions, then maybe we could return `None`  to let processing loop know that not a single object has been detected and that it should return 204 for this request?